pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -277,6 +277,7 @@ func (c *Controller) initializeOperation(ctx context.Context, logger *logrus.Ent
 		WithCloudProfileObject(cloudProfile).
 		WithShootSecretFromReader(gardenClient.APIReader()).
 		WithProjectName(project.Name).
+		WithExposureClassFromReader(gardenClient.APIReader()).
```

Comment:
You need to extend the seed authorizer, similar to what @danielfoehrKn does as part of #3893 (changes in `pkg/admissioncontroller`). As the gardenlet only reads the `ExposureClass`es, your implementation will be easier and you don't need to enhance the `seedrestriction` admission plugin.